### ISSUE TITLE
performance on AMD 390x

### ISSUE THREAD

**Issue by manotroll** [yuzu-emu/yuzu#2652]
why performance is / is still bad on amd cards?
Is there any way to improve performance?
is there any way to force a full disk shader creation before starting the game?
where does the disk shader get saved?
Is it possible to add a shortcut in the GUI to open the shader directory of the running game?

**Comment by BSoDGamingYT** [comment]
Why performance is / is still bad on amd cards?
AMD windows drivers for OpenGL applications is REALLY bad.

Is there any way to improve performance?
You can try to use Linux with the MESA driver. You will have significantly better performance there. 

Is there any way to force a full disk shader creation before starting the game?
No

Where does the disk shader get saved?
AppData\Roaming\yuzu\shader\opengl\transferable

Is it possible to add a shortcut in the GUI to open the shader directory of the running game?
This is already a part of the GUI. Simply right-click your game in the games list & select "Open Transferable shader cache" If a shader cache exists for the game, this will open its location. To make one, simply play the game

**Comment by manotroll** [comment]
thanks then the performance would be better on linux in which version should i use which distro?
MESA driver would be an open source driver or use the owner?
the difference and significant to compensate for a dual boot?
vulkan comes end of the year for everyone?

**Comment by BSoDGamingYT** [comment]
This is not a place for discussions like this. If you want to talk about yuzu casually. Join their discord

**Comment by FernandoS27** [comment]
Sadly the AMD driver for windows is terrible. We have tried everything to improve it but to no avail.

**Comment by manotroll** [comment]
it's open for a long time and only created 2 files in the shader running at 8 fps so waiting to see if it creates more and gains speed

**Comment by FernandoS27** [comment]
Have you tried homebrew games? Those could help give some info on the problem.

**Comment by Socke81** [comment]
_Is there any way to force a full disk shader creation before starting the game?_

Could a developer (please no answer from a Yutuber) explain this in more detail or post a link why? This also applies to other emulators and I've been wondering why for a long time.

**Comment by manotroll** [comment]
other games work
is that I'm trying to play Crash Team Racing Nitro-Fueled since it had taken 1 year to leave for pc
it does the animation at the beginning and then freezes

**Comment by ReinUsesLisp** [comment]
@Socke81 Most games ship precompiled shaders. This means that they are in a binary format host (your PC) can't execute, even if it's an Nvidia GPU (due to some instructions and edge cases not supported on desktop). We don't know where these shaders are stored, and some might even be generated (or patched) at runtime by the guest (game); our most reliable option is to grab these shaders when the game is rendering, at this moment we know exactly when the shader is. This generates the stutter you might be familiar with. This is why most modern emulators provide a disk shader cache mechanism.

**Comment by Socke81** [comment]
Yes, but when the shader is found, you have the binary. With the binary data you could now find it within the game data. Of course not in real time but by option. If the binary is found in the game data then you could save the position. You can publish the position with yuzu because there is no copyright on it. This would allow the next user to compile the shaders before starting the game. Of course, this does not affect the self-changing shaders. But how many of them are there compared to the normal ones?

**Comment by ReinUsesLisp** [comment]
We don't know where the data comes from. We know it's address at runtime, but not where it is in the game's static storage.

**Comment by Socke81** [comment]
Yeah, I know,. That's why I also wrote it should be searched in. You take a few bytes (or everything, I don't know how big they are) from the shader and search for them in the game data.

**Comment by ReinUsesLisp** [comment]
@Socke81 These could be encrypted/compressed/encoded in proprietary formats. It is theoretically possible, but it's a non-issue having a disk shader cache. I invite you to discuss this in a separate issue or in Discord.

**Comment by ReinUsesLisp** [comment]
Closing this as wont-fix. AMD's proprietary OpenGL driver's performance is awful and we can't fix it from yuzu's end.
